tests/test_synchronized.py

```diff
@@ -0,0 +1,199 @@
+    assert not producer.is_consumers_queue_empty(1)
+    assert not producer.is_consumers_queue_empty(2)
+    await producer.synchronized_perform_consumers_queue(1)
+    assert producer.is_consumers_queue_empty(1)
+    assert producer.is_consumers_queue_empty(2)
+
+
+@pytest.mark.asyncio
+async def test_is_consumers_queue_empty_with_multiple_consumers(synchronized_channel):
+    async def callback():
+        pass
+
+    await synchronized_channel.new_consumer(callback)
+    await synchronized_channel.new_consumer(callback)
+    await synchronized_channel.new_consumer(callback, priority_level=2)
+    await synchronized_channel.new_consumer(callback, priority_level=2)
+    await synchronized_channel.new_consumer(callback, priority_level=3)
+
+    producer = TestSynchronizedProducer(get_chan(TEST_SYNCHRONIZED_CHANNEL))
+    await producer.run()
+
+    await producer.send({})
+    assert not producer.is_consumers_queue_empty(1)
+    assert not producer.is_consumers_queue_empty(2)
+    assert not producer.is_consumers_queue_empty(3)
+    await producer.synchronized_perform_consumers_queue(1)
+    assert producer.is_consumers_queue_empty(1)
+    assert not producer.is_consumers_queue_empty(2)
+    assert not producer.is_consumers_queue_empty(3)
+    await producer.synchronized_perform_consumers_queue(2)
+    assert producer.is_consumers_queue_empty(1)
+    assert producer.is_consumers_queue_empty(2)
+    assert not producer.is_consumers_queue_empty(3)
+    await producer.synchronized_perform_consumers_queue(2)
+    assert not producer.is_consumers_queue_empty(3)
+    await producer.synchronized_perform_consumers_queue(3)
+    assert producer.is_consumers_queue_empty(3)
+
+
+@pytest.mark.asyncio
+async def test_producer_synchronized_perform_consumers_queue_with_multiple_consumer(synchronized_channel):
+    async def callback():
+        pass
+
+    test_consumer_1_1 = await synchronized_channel.new_consumer(callback)
+    test_consumer_1_2 = await synchronized_channel.new_consumer(callback)
+    test_consumer_2_1 = await synchronized_channel.new_consumer(callback, priority_level=2)
+    test_consumer_2_2 = await synchronized_channel.new_consumer(callback, priority_level=2)
+    test_consumer_3_1 = await synchronized_channel.new_consumer(callback, priority_level=3)
+
+    producer = TestSynchronizedProducer(get_chan(TEST_SYNCHRONIZED_CHANNEL))
+    await producer.run()
+
+    with patch.object(test_consumer_1_1, 'callback', new=AsyncMock()) as mocked_test_consumer_1_1_callback:
+        with patch.object(test_consumer_1_2, 'callback', new=AsyncMock()) as mocked_test_consumer_1_2_callback:
+            with patch.object(test_consumer_2_1, 'callback', new=AsyncMock()) as mocked_test_consumer_2_1_callback:
+                with patch.object(test_consumer_2_2, 'callback', new=AsyncMock()) as mocked_test_consumer_2_2_callback:
+                    with patch.object(test_consumer_3_1, 'callback', new=AsyncMock()) as mocked_test_consumer_3_1_callback:
```

Comment:
you can chain multiple context managers with a "," 😃

Reply:
Is it possible ? Great !